R/Rules-class.R

```diff
@@ -139,13 +169,101 @@ validObject(.NextBestNCRM())
 ##' @keywords methods
 NextBestNCRM <- function(target,
                          overdose,
-                         maxOverdoseProb)
-{
-    .NextBestNCRM(target=target,
-                  overdose=overdose,
-                  maxOverdoseProb=maxOverdoseProb)
+                         maxOverdoseProb) {
+  .NextBestNCRM(
+    target = target,
+    overdose = overdose,
+    maxOverdoseProb = maxOverdoseProb
+  )
 }
 
+##' Next best dose based on NCRM rule and loss function
+##'
+##' This class is similar to NCRM class, the only difference is the addition of
+##' loss function. As in NCRM rule, first admissible doses are found, which are those
+##' with probability to fall in \code{overdose} category being below
+##' \code{maxOverdoseProb}. Next, within the admissible doses, the loss function is
+##' calculated, i.e. \code{losses} %*% \code{target}. Finally, the corresponding
+##' dose with lowest loss function (Bayes risk) is recommended for the next dose.
+##' @slot target the target toxicity interval (limits included)
+##' @slot overdose the overdose toxicity interval (lower limit excluded, upper limit included)
+##' @slot maxOverdoseProb maximum overdose probability that is allowed
+##' @slot losses loss function
+##'
+##' @example examples/Rules-class-NextBestNCRMLoss.R
+##' @export
+##' @keywords classes
+.NextBestNCRMLoss <-
+  setClass(
+    Class = "NextBestNCRMLoss",
+    representation(
+      target = "numeric",
+      overdose = "numeric",
+      maxOverdoseProb = "numeric",
+      losses = "numeric"
+    ),
+    prototype(
+      target = c(0.2, 0.35),
+      overdose = c(0.35, 0.6, 1),
```

Comment:
why does this have 3 elements? I guess to have 2 intervals here?
then I think it would be better to have implicit definition of unacceptable toxicity as anything above `overdose[2]` and reduce this to `overdose = c(0.35, 0.6)`

Reply:
Yes, code is updated as suggested